client/src/components/sources/sources.js

```diff
@@ -148,13 +152,37 @@ class Sources extends React.Component {
     });
   }
 
-  addSource() {
+  showAddSourceWizard() {
+    this.setState({ addSourceWizardShown: true });
+
+    /*
     Store.dispatch({
       type: toastNotificationTypes.TOAST_ADD,
       alertType: 'error',
       header: 'NYI',
       message: 'Adding sources is not yet implemented'
     });
+    */
+  }
+
+  quitAddSourceWizard() {
+    let heading = <h3>Are you sure you want to quit adding a source?</h3>;
+
+    let onConfirm = () => {
+      Store.dispatch({
+        type: confirmationModalTypes.CONFIRMATION_MODAL_HIDE
+      });
+
+      this.setState({ addSourceWizardShown: false });
+    };
+
+    Store.dispatch({
+      type: confirmationModalTypes.CONFIRMATION_MODAL_SHOW,
+      title: 'Quit Adding Source',
```

Comment:
How about 'Cancel Add Source'?

Reply:
heh so i started to use the ```cancel``` wording except... there's a "cancel" button on the confirmation modal... that feels like a dark pattern...